app/models/user.rb

```diff
@@ -637,6 +638,12 @@ def admin_username_generic
     errors.add(:login, :admin_must_use_default) unless login == "user#{id}"
   end
 
+  def username_must_be_different_from_current
+    return if User.current_user != self
+
+    errors.add(:login, :username_not_different) if login == login_was
```

Comment:
Nitpick: this can just be
```suggestion
    errors.add(:login, :not_different) if login == login_was
```
since the error is already scoped to `:login`